packages/amplify-cli-core/src/hooks/hooksMeta.ts

```diff
@@ -0,0 +1,155 @@
+import { HookEvent, DataParameter, EventPrefix, HooksVerb, HooksNoun, ErrorParameter } from './hooksTypes';
+import { suppportedEvents, supportedEnvEvents } from './hooksConstants';
+import { stateManager } from '../state-manager';
+import _ from 'lodash';
+
+export class HooksMeta {
+  private static instance?: HooksMeta;
+  private HookEvent: Partial<HookEvent>;
+  private dataParameter: DataParameter;
+  private errorParameter: ErrorParameter | undefined;
+
+  public static getInstance = (HookEvent: Partial<HookEvent> = {}, dataParameter: DataParameter = { amplify: {} }): HooksMeta => {
```

Comment:
is there any case where this should be called before `constructHooksMetaObject`? If not, I would suggest modifying this method to throw if not initialized. This makes it clear that `constructHooksMetaObject` (which I propose renaming to `initialize`) must be called before an instance can be returned.

My concern here is that there is some interplay between `constructHooksMetaObject` and `getInstance` that is not obvious just from the interface. If we can make it obvious that `constructHooksMetaObject` must be called first, that would be ideal

Regardless of what you decide above, it doesn't looks like the input params are supplied anywhere so I would remove them.

Reply:
done